### ISSUE TITLE
Help on Mailserverver Tutorial

### ISSUE THREAD

**Issue by Elleni57** [davidmainzer/davidmainzer.github.io#1]
Hi David Mainzer, 

Following your guide Mailserver for gentoo to the point: 

Compile Sieve files: will generate *.svbin files
sievec FILE.sieve 

I get: doveconf: Fatal: Error in configuration file /etc/dovecot/conf.d/90-sieve.conf line 210: Unknown setting: imapsieve_mailbox1_name

**Comment by davidmainzer** [comment]
Hi,

can you provide the complete command line incl output?

I used all these steps to setup my mailserver one year ago.

Best,
dm

**Comment by Elleni57** [comment]
Hi, 

in the meantime I have setup postfix/dovecot with dkim amavis-new clamav and spamassassin successfully but with other database-, username, mailbox location. 

I will try to adapt the steps described in your howto to setup rspamd and come back if the problem persists. 

Thanks

**Comment by Elleni57** [comment]
Hello, 

0k, so I am trying again. I am not sure, where in /etc/dovecot/conf.d/90-sieve.conf the following has to be put: 
#Spam learning
#From elsewhere to Spam folder
  imapsieve_mailbox1_name = Spam
  imapsieve_mailbox1_causes = COPY
  imapsieve_mailbox1_before = file:/var/mail/sieve/global/learn-spam.sieve

#From Spam folder to elsewhere
  imapsieve_mailbox2_name = *
  imapsieve_mailbox2_from = Spam
  imapsieve_mailbox2_causes = COPY
  imapsieve_mailbox2_before = file:/var/mail/sieve/global/learn-ham.sieve

  sieve_pipe_bin_dir = /usr/bin
  sieve_global_extensions = +vnd.dovecot.pipe

Putting it outside plugin {...}, the initially reported error has occured. 

But putting it inside plugin {..}, there are still errors while executing 
sievec. 

If I understand you correctly I have to go to global folder, which in my case is 
cd /var/mail/sieve/global/

and execute from there: 

sievec spam-global.sieve
sievec(root): Warning: sieve: ignored unknown extension 'vnd.dovecot.pipe' while configuring available extensions

sievec learn-ham.sieve 
sievec(root): Warning: sieve: ignored unknown extension 'vnd.dovecot.pipe' while configuring available extensions
learn-ham: line 1: error: require command: unknown Sieve capability `vnd.dovecot.pipe'.
learn-ham: line 1: error: require command: unknown Sieve capability `imapsieve'.
learn-ham: line 2: error: unknown command 'pipe' (only reported once at first occurrence).
learn-ham: error: validation failed.
sievec(root): Fatal: failed to compile sieve script 'learn-ham.sieve'

sievec learn-spam.sieve 
sievec(root): Warning: sieve: ignored unknown extension 'vnd.dovecot.pipe' while configuring available extensions
learn-spam: line 1: error: require command: unknown Sieve capability `vnd.dovecot.pipe'.
learn-spam: line 1: error: require command: unknown Sieve capability `imapsieve'.
learn-spam: line 2: error: unknown command 'pipe' (only reported once at first occurrence).
learn-spam: error: validation failed.
sievec(root): Fatal: failed to compile sieve script 'learn-spam.sieve' 

What am I missing?

**Comment by davidmainzer** [comment]
Do you have pigeonhole extension for dovecot?

locate pigeonhole

i think 'managesieve' use flag for dovecot is important. Here are my dovecot use flags:

[I] net-mail/dovecot
     Available versions:  ~2.2.36.4 2.3.7.2 ~2.3.8 {argon2 bzip2 caps doc ipv6 kerberos ldap libressl lua lucene lz4 lzma managesieve mysql pam postgres selinux sieve solr sqlite static-libs suid tcpd textcat vpopmail zlib}
     Installed versions:  2.3.7.2(10:39:44 PM 10/22/2019)(bzip2 ipv6 managesieve mysql pam sieve tcpd zlib -argon2 -caps -doc -kerberos -ldap -libressl -lua -lucene -lz4 -lzma -postgres -selinux -solr -sqlite -static-libs -suid -textcat -vpopmail)
     Homepage:            https://www.dovecot.org/
     Description:         An IMAP and POP3 server written with security primarily in mind


Are u using a gentoo?

Please check your installation. 

https://forum.iredmail.org/topic9696-iredmail-support-sieve-extprogram-plugin-is-it-really-included-in-this-release.html

**Comment by Elleni57** [comment]
yes, I am using a freshly installed and up to date gentoo. 

locate pigeonhole
/usr/include/dovecot/sieve/pigeonhole-config.h
/usr/include/dovecot/sieve/pigeonhole-version.h
/usr/lib64/dovecot/settings/libpigeonhole_settings.so
/usr/share/aclocal/dovecot-pigeonhole.m4
/usr/share/man/man7/pigeonhole.7.bz2

managesieve and sieve useflags are already set. 

hostname ~ # emerge dovecot -pv

These are the packages that would be merged, in order:

Calculating dependencies        ... done!                
[ebuild   R    ] net-mail/dovecot-2.3.7.2::gentoo  USE="bzip2 caps managesieve mysql pam sieve sqlite tcpd zlib -argon2 -doc -ipv6 -kerberos -ldap -libressl -lua -lucene -lz4 -lzma -postgres (-selinux) -solr -static-libs -suid -textcat -vpopmail" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

uname -a
Linux itsrv2 4.19.82-gentoo #2 SMP Tue Nov 12 01:07:15 CET 2019 x86_64 AMD EPYC Processor (with IBPB) AuthenticAMD GNU/Linux

emerge --info
Portage 2.3.76 (python 3.6.9-final-0, default/linux/amd64/17.1/no-multilib, gcc-9.2.0, glibc-2.29-r2, 4.19.82-gentoo x86_64)
System uname: Linux-4.19.82-gentoo-x86_64-AMD_EPYC_Processor_-with_IBPB-with-gentoo-2.6
KiB Mem:     6103940 total,   3041472 free
KiB Swap:    6291452 total,   6286332 free
Timestamp of repository gentoo: Fri, 15 Nov 2019 16:30:01 +0000
Head commit of repository gentoo: eb8dbe94cae643c4db4fe1842ec03c6b5b607dee
sh bash 4.4_p23-r1
ld GNU ld (Gentoo 2.32 p2) 2.32.0
app-shells/bash:          4.4_p23-r1::gentoo
dev-lang/perl:            5.28.2-r1::gentoo
dev-lang/python:          2.7.16::gentoo, 3.6.9::gentoo
dev-util/cmake:           3.14.6::gentoo
sys-apps/baselayout:      2.6-r1::gentoo
sys-apps/openrc:          0.41.2::gentoo
sys-apps/sandbox:         2.13::gentoo
sys-devel/autoconf:       2.69-r4::gentoo
sys-devel/automake:       1.16.1-r1::gentoo
sys-devel/binutils:       2.32-r1::gentoo
sys-devel/gcc:            9.2.0-r2::gentoo
sys-devel/gcc-config:     2.1::gentoo
sys-devel/libtool:        2.4.6-r3::gentoo
sys-devel/make:           4.2.1-r4::gentoo
sys-kernel/linux-headers: 4.19::gentoo (virtual/os-headers)
sys-libs/glibc:           2.29-r2::gentoo
Repositories:

gentoo
    location: /var/db/repos/gentoo
    sync-type: rsync
    sync-uri: rsync://rsync.gentoo.org/gentoo-portage
    priority: -1000
    sync-rsync-verify-jobs: 1
    sync-rsync-extra-opts: 
    sync-rsync-verify-max-age: 24
    sync-rsync-verify-metamanifest: yes

local_overlay
    location: /usr/local/portage
    masters: gentoo
    priority: 0

ACCEPT_KEYWORDS="amd64"
ACCEPT_LICENSE="@FREE"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=native -O2 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/easy-rsa /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/php/apache2-php7.3/ext-active/ /etc/php/cgi-php7.3/ext-active/ /etc/php/cli-php7.3/ext-active/ /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo"
CXXFLAGS="-march=native -O2 -pipe"
DISTDIR="/var/cache/distfiles"
ENV_UNSET="DBUS_SESSION_BUS_ADDRESS DISPLAY GOBIN PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR"
FCFLAGS="-march=native -O2 -pipe"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs candy config-protect-if-modified distlocks ebuild-locks fixlafiles ipc-sandbox merge-sync multilib-strict network-sandbox news parallel-fetch pid-sandbox preserve-libs protect-owned sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-march=native -O2 -pipe"
GENTOO_MIRRORS="http://distfiles.gentoo.org"
LANG="de_CH.UTF-8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
LINGUAS="de de_DE el en fr fr_FR it tr"
MAKEOPTS="-j5"
PKGDIR="/var/cache/binpkgs"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
USE="3dnow 3dnowext acl amd64 apache2 authdaemond berkdb bzip2 caps cgi clamav clamdtop cli crypt cryptsetup curl cxx device-mapper dkim dovecot-sasl dri exif fam fontconfig fortran fpm gd gdbm geoip iconv imap jpeg libmysqlclient libtirpc maildir managesieve mmx mmxext mysql mysqli ncurses nls nptl openmp pam pcntl pcre pdo png popcnt readline seccomp sieve sockets spamassassin spell split-usr sqlite sse sse2 sse3 sse4_1 sse4a ssl symlink tcpd truetype unicode vhosts xattr xmlwriter xslt zip zlib" ABI_X86="64" ADA_TARGET="gnat_2018" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="actions alias auth_basic authn_alias authn_anon authn_core authn_dbm authn_file authz_core authz_dbm authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate env expires ext_filter file_cache filter headers include info log_config logio mime mime_magic negotiation proxy proxy_http proxy_wstunnel rewrite setenvif socache_shmcb speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 f16c fma3 mmx mmxext pclmul popcnt sha sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" INPUT_DEVICES="libinput keyboard mouse" KERNEL="linux" L10N="de el en fr it tr" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" NETBEANS_MODULES="apisupport cnd groovy gsf harness ide identity j2ee java mobility nb php profiler soa visualweb webcommon websvccommon xml" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php7-3" POSTGRES_TARGETS="postgres10 postgres11" PYTHON_SINGLE_TARGET="python3_6" PYTHON_TARGETS="python2_7 python3_6" RUBY_TARGETS="ruby24 ruby25" USERLAND="GNU" VIDEO_CARDS="amdgpu fbdev intel nouveau radeon radeonsi vesa dummy v4l" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, EMERGE_DEFAULT_OPTS, INSTALL_MASK, LC_ALL, PORTAGE_BINHOST, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS

**Comment by Elleni57** [comment]
I think I found the problem, comparing Thomas Leisters article. 

This is missing in yours: 
    sieve_plugins = sieve_imapsieve sieve_extprograms
    sieve_before = /var/mail/sieve/global/spam-global.sieve
    sieve = file:/var/mail/sieve/%d/%n/scripts;active=/var/mail/sieve/%d/%n/active-script.sieve

**Comment by Elleni57** [comment]
Almost there. I finished configuring the system. Now I still cant send mails. Checking /var/log/messages I see: 
postfix/smtpd[3732]: fatal: host/service localhost/11332 not found: No address associated with hostnamepostfix/smtpd[3732]: fatal: host/service localhost/11332 not found: No address associated with hostname
Any idea how to correct?

**Comment by Elleni57** [comment]
As i was able to finish setup and everything works fine so sofar, I want to thank you. 

Other than that, I stumbeled over the problem that when mails are being deleted from spam they would be relearned as ham. I found a quite simple solution which I want to share, so maybe you want to add it to your guide. 

See last post in my gentoo thread: 
https://forums.gentoo.org/viewtopic-p-8391700.html#8391700
